### PR TITLE
#1794: Allow Guests to browse without SlackId

### LINKED ISSUE #1794: Users Should Not Be Forced to Enter Their Slack ID as Guests
### Description

Currently, whenever anyone logs into finishline for the first time, they get an input box, prompting them to enter their slack id. We neither want nor need a guests slack id. It would be better to allow someone (a new member or someone we're showing the site to) to look around the site as a guest in 'View Only' mode and then only require slack ids for anyone who has been upgraded to a member or higher. 

### Acceptance Criteria

- [ ] The prompt making a user enter their slack id when they first open FinishLine only appears when the user is not a guest.

### Proposed Solution

Poke around the code and find out where that prompt actually is. From there, there are plenty of references for not allowing a guest to do something that you can reference in our frontend code.

### PR DESCRIPTION
**Pull request by Dao-Ho**
## Changes

Passed in userRole into AppAuthenticated along with existing userId, allowing us to check whether the user is a guest. If the user is a guest, allow them to continue to the site without the SlackId requirement. Otherwise, prompt the requirement if they are a member, lead, admin, etc. Also removed the alert for guests to fill out their slack ID in the home page. 

Also removed Slack alert to set Slack ID for all users in the home page (as well as unnecessary imports due to this removal).

## Notes

Also added userRole as a parameter for AppAuthenticated.test.tsx when testing the component.

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1794

### CODE REVIEW

**Review by callmejim1226 — CHANGES_REQUESTED** [review]
Great work! The current code looks good to me :)

Although, I think it would be better to make a further update. Curretnly, when we log in as a GUEST in Finishline, we still see a yellow warning pop-up in the top as the screenshot shows. 

Can we remove this pop-up when the user is logged into Finishline as a guest?

image

File: yarn.lock
Comment (callmejim1226) [inline yarn.lock:1]: We don't want to have any changes in yarn.lock file unless the project's dependencies have changed (which is not the case in this commit). 

Can we undo the changes in this yarn.lock file?

**Review by callmejim1226 — CHANGES_REQUESTED** [review]
Great work! Almost there! Just a single comment on yarn.lock file since we don't any changes in yarn.lock to be pushed into develop branch

**Review by callmejim1226 — APPROVED** [review]
Looks good to me! Great work!

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**